datatypes-jws/src/commonMain/kotlin/at/asitplus/crypto/datatypes/jws/JsonWebKey.kt

```diff
@@ -65,18 +75,16 @@ data class JsonWebKey(
         result = 31 * result + (keyId?.hashCode() ?: 0)
         result = 31 * result + (x?.contentHashCode() ?: 0)
         result = 31 * result + (y?.contentHashCode() ?: 0)
+        result = 31 * result + (n?.hashCode() ?: 0)
+        result = 31 * result + (e?.hashCode() ?: 0)
         return result
-
-        //todo n, e
     }
 
     companion object {
 
         fun deserialize(it: String) = kotlin.runCatching {
             jsonSerializer.decodeFromString<JsonWebKey>(it)
-        }.getOrElse {
-            null
-        }
+        }.getOrNull()
```

Comment:
Maybe ... but then wrapped in a `KmmResult` because it could get called from Swift?

Reply:
Every parsing throws errors all over the place. we should annotate it and document it, but not hide it IMHO.
What's the deal with Swift and exceptions again?

Reply:
We'll just need to mark it as `@Throws` with the correct list of exceptions, because every other exception is unchecked, and Swift only has checked errors, so anything unexpected causes program termination, see [Errors and exceptions](https://kotlinlang.org/docs/native-objc-interop.html#errors-and-exceptions). But that's probably best handled in a new issue.